homeassistant/components/binary_sensor/deconz.py

```diff
@@ -0,0 +1,91 @@
+"""
+Support for deCONZ binary sensor support.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/binary_sensor/deconz/
+"""
+
+import asyncio
+import logging
+
+from homeassistant.components.binary_sensor import BinarySensorDevice
+from homeassistant.components.deconz import (
+    ATTR_FW_VERSION, ATTR_MANUFACTURER, ATTR_MODEL_ID, ATTR_UNIQUE_ID,
+    DECONZ_DATA, DOMAIN)
+from homeassistant.const import ATTR_BATTERY_LEVEL
+from homeassistant.core import callback
+
+DEPENDENCIES = [DOMAIN]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@asyncio.coroutine
+def async_setup_platform(hass, config, async_add_devices, discovery_info=None):
+    """Setup binary sensor platform for Deconz."""
+    from pydeconz.sensor import DECONZ_BINARY_SENSOR
+    if DECONZ_DATA in hass.data:
+        sensors = hass.data[DECONZ_DATA].sensors
+
+    for _, sensor in sensors.items():
```

Comment:
Should probably be inside of the above `if` statement, otherwise `sensors` will be undefined.

Reply:
Sensors are initiated as a dict so not an issue.

Reply:
Only if `hass.data[DECONZ_DATA]` could be found (then it should be an instance of `DeconzSession),` otherwise the variable will not be set at all here. So maybe either remove the above `if DECONZ_DATA in hass.data:` completely and rely on it being set, or move the `for` loops into the `if` statement.

Or am I missing something here?

Reply:
Aah, sure. Yes, my plan is to remove the if statement